lib/WebpackOptionsDefaulter.js

```diff
@@ -267,7 +267,12 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 					new UglifyJsPlugin({
 						cache: true,
 						parallel: true,
-						sourceMap: options.devtool && /source-?map/.test(options.devtool)
+						sourceMap:
+							(options.devtool && /source-?map/.test(options.devtool)) ||
+							(options.plugins &&
+								options.plugins.some(
+									p => p instanceof require("./SourceMapDevToolPlugin")
```

Comment:
for those scared of multiple require calls to the same file in the loop here, from nodejs doc: "Modules are cached after the first time they are loaded. This means (among other things) that every call to require('foo') will get exactly the same object returned, if it would resolve to the same file."

Reply:
Yeah technically it's fine, but better move the require next to the uglifyjs-webpack-plugin require.

Reply:
Yeah, require is cached. I put it here as opposed to by the above make it more lazy-loaded. I guess it's pretty minimal, I'll go ahead and move it next to uglify.